internal/compiler/program.go

```diff
@@ -331,8 +342,479 @@ func (p *Program) GetSuggestionDiagnostics(ctx context.Context, sourceFile *ast.
+		createOptionValueDiagnostic("allowImportingTsExtensions", diagnostics.Option_allowImportingTsExtensions_can_only_be_used_when_either_noEmit_or_emitDeclarationOnly_is_set)
+	}
+
+	moduleResolution := options.GetModuleResolutionKind()
+	if options.ResolvePackageJsonExports.IsTrue() && !moduleResolutionSupportsPackageJsonExportsAndImports(moduleResolution) {
+		createDiagnosticForOptionName(diagnostics.Option_0_can_only_be_used_when_moduleResolution_is_set_to_node16_nodenext_or_bundler, "resolvePackageJsonExports", "")
+	}
+	if options.ResolvePackageJsonImports.IsTrue() && !moduleResolutionSupportsPackageJsonExportsAndImports(moduleResolution) {
+		createDiagnosticForOptionName(diagnostics.Option_0_can_only_be_used_when_moduleResolution_is_set_to_node16_nodenext_or_bundler, "resolvePackageJsonImports", "")
+	}
+	if options.CustomConditions != nil && !moduleResolutionSupportsPackageJsonExportsAndImports(moduleResolution) {
+		createDiagnosticForOptionName(diagnostics.Option_0_can_only_be_used_when_moduleResolution_is_set_to_node16_nodenext_or_bundler, "customConditions", "")
+	}
+
+	// !!! Reenable once we don't map old moduleResolution kinds to bundler.
+	// if moduleResolution == core.ModuleResolutionKindBundler && !emitModuleKindIsNonNodeESM(moduleKind) && moduleKind != core.ModuleKindPreserve {
+	// 	createOptionValueDiagnostic("moduleResolution", diagnostics.Option_0_can_only_be_used_when_module_is_set_to_preserve_or_to_es2015_or_later, "bundler")
+	// }
+
+	if core.ModuleKindNode16 <= moduleKind && moduleKind <= core.ModuleKindNodeNext &&
+		!(core.ModuleResolutionKindNode16 <= moduleResolution && moduleResolution <= core.ModuleResolutionKindNodeNext) {
+		moduleKindName := moduleKind.String()
+		var moduleResolutionName string
+		if v, ok := core.ModuleKindToModuleResolutionKind[moduleKind]; ok {
+			moduleResolutionName = v.String()
+		} else {
+			moduleResolutionName = "Node16"
+		}
+		createOptionValueDiagnostic("moduleResolution", diagnostics.Option_moduleResolution_must_be_set_to_0_or_left_unspecified_when_option_module_is_set_to_1, moduleResolutionName, moduleKindName)
+	} else if core.ModuleResolutionKindNode16 <= moduleResolution && moduleResolution <= core.ModuleResolutionKindNodeNext &&
+		!(core.ModuleKindNode16 <= moduleKind && moduleKind <= core.ModuleKindNodeNext) {
+		moduleResolutionName := moduleResolution.String()
+		createOptionValueDiagnostic("module", diagnostics.Option_module_must_be_set_to_0_when_option_moduleResolution_is_set_to_1, moduleResolutionName, moduleResolutionName)
+	}
+
+	// If the emit is enabled make sure that every output file is unique and not overwriting any of the input files
+	// if !options.NoEmit.IsTrue() && !options.SuppressOutputPathCheck.IsTrue() {
+	// 	var emitFilesSeen collections.Set[string]
+
+	// 	// Verify that all the emit files are unique and don't overwrite input files
+	// 	verifyEmitFilePath := func(emitFileName string) {
```

Comment:
Comment on why this block is commented out

Reply:
Rechecking this now; I forgot what broke.